File: 04_Microbot/Python/controller_base.py
#robot-ctrl.py
#Robot controller

from microbit import *
import radio

radio_group = 0 #number of radio frequency to use
forward = False
group_set = False #has the radio group been set?
radio.on()
radio.config(power=7)

stop_sign = Image(  "09990:"
                    "90099:"
                    "90909:"
                    "99009:"
                    "09990")

while True:# do forever
        if group_set is False:
            # Show radio_group on the display (use str(radio_group))
            

        if button_a.is_pressed() and button_b.is_pressed():
            if group_set is False:
                # Set group_set to True
                
                radio.config(channel = radio_group)
                # Show a happy face on the display (Image.HAPPY)
                
                # Set forward to False
            else:
                if forward == False:
                    # Send "forward" over radio (radio.send(...))
                    
                    # Set forward to True

                    # Show a forward arrow on the display (Image.ARROW_N)
                    
                else:
                    # Send "stop" over radio (radio.send(...))
                    
                    # Set forward to False

                    # Show a stop sign on the display (stop_sign)
            
        elif button_a.is_pressed():
            if group_set == False and radio_group > 0 :
                # Decrement radio_group by one (radio_group-1)
                
            else:
                # Send "left" over radio

                # Show a left arrow on the display (Image.ARROW_W)
                
        elif button_b.is_pressed():
            if group_set is False and radio_group < 255:
                # Increment radio_group by one (radio_group+1)

            else:
                # Send "right" over radio

                # Show a right arrow on the display (Image.ARROW_E)
        sleep(200)

File: 04_Microbot/Javascript/controller_base.js
//NEED TO UPDATE
let forward = 0
let groupSet = 0
let radioGroup = 0
input.onButtonPressed(Button.A, () => {
    if (groupSet == 0 && radioGroup > 0) {
        // Decrement radioGroup by 1 (radioGroup-1)

    } else {
        // Send "left" over radio (radio.sendString(...))

        basic.showLeds(`
            . . # . .
            . # . . .
            # # # # #
            . # . . .
            . . # . .
            `)
    }
})
input.onButtonPressed(Button.AB, () => {
    if (groupSet == 0) {
        // Set grouopSet to 1

        radio.setGroup(radioGroup)
        // Show a happy face on the display (IconNames.Happy)

        // Set forward to zero
    } else {
        if (forward == 0) {
            // Send "forward" over radio

            // Set forward to one

            basic.showLeds(`
                . . # . .
                . # # # .
                # . # . #
                . . # . .
                . . # . .
                `)
        } else {
            // Send "stop" over radio


            // Set forward to zero

            basic.showLeds(`
                . # # # .
                # . . # #
                # . # . #
                # # . . #
                . # # # .
                `)
        }
    }
})
input.onButtonPressed(Button.B, () => {
    if (groupSet == 0 && radioGroup < 255) {
        // Increment radioGroup by 1 (radioGroup+1)

    } else {
        // Send "right" over radio

        basic.showLeds(`
            . . # . .
            . . . # .
            # # # # #
            . . . # .
            . . # . .
            `)
    }
})
radioGroup = 0
while (groupSet != 1) {
    basic.showNumber(radioGroup)
}
basic.showIcon(IconNames.Happy)
basic.pause(100)
basic.showNumber(radioGroup)
